changelog/unreleased/10557

```diff
@@ -0,0 +1,6 @@
+Enhancement: Improve display of connection state
+
+Don't display `No connection configured` when disconnected but `Disconnected from:`.
+Introduce new state, `Connecting to`.
```

Comment:
We split the current "no connection configured" state into a "disconnected" and "connecting to" state to better explain the client's behavior to the user.